web/src/ApiButton.tsx

```diff
@@ -19,6 +42,97 @@ const svgElement = (src: string): React.ReactElement => {
   return convertFromNode(node) as React.ReactElement
 }
 
+type ApiButtonInputProps = {
+  spec: UIInputSpec
+  status: UIInputStatus | undefined
+  value: string | undefined
+  setValue: (name: string, value: string) => void
+}
+
+function ApiButtonInput(props: ApiButtonInputProps) {
```

Comment:
i'm surprised you didn't inline this; i'm not sure it's adding much value as an independent component.